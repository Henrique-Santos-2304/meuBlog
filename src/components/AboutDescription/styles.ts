import styled, { css } from "styled-components";
export const ContentAbout = styled.div`
  height: max-content;
  padding: 1.8rem 0;
`;
export const TextDescription = styled.p`
  ${({ theme }) => css`
    margin-top: 1rem;
    font-size: ${theme.font.sizes.medium};
    font-weight: 500;

    strong {
      font-style: italic;
    }
  `}
`;
