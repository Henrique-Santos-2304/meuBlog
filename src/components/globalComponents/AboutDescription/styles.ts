import styled, { css } from "styled-components";
export const ContentAbout = styled.div`
  height: max-content;
  padding: 1.8rem 0;
`;
export const TextDescription = styled.p`
  ${({ theme }) => css`
    margin-top: 2rem;
    font-size: ${theme.font.sizes.small};
    font-weight: 500;

    strong {
      font-style: italic;
    }
  `}
`;
