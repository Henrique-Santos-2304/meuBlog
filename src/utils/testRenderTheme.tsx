import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
