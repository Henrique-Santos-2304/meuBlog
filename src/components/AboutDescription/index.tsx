import Headings from "components/globalComponents/Headings";
import SectionClipPathEffect from "components/globalComponents/SectionClipPathEffect";
import * as S from "./styles";

const AboutDescription = () => (
  <SectionClipPathEffect>
    <S.ContentAbout>
      <Headings
        msg="Um pouco sobre mim"
        colorText="#1e1e1e"
        colorEffect="#486afe"
        fontSizeText="2rem"
        marginText="0 0 1rem 1.5rem"
        aria-label="Titulo da seção"
      />
      <S.TextDescription aria-label="Texto com a descrição do Programador">
        Sou um Desenvolvedor a princípio Front End, com foco em
        <strong> React | Next.js</strong>, apaixonado pela Programação estou em
        constante aprendizado e desenvolvimento profissional sempre me
        atualizando, no momento crio projetos Pesssoais e freelances voltados a
        Desenvolvimento Web, mas ao mesmo tempo estudando sobre desenvolvimento
        Mobile com <strong>React Native</strong> .
      </S.TextDescription>
    </S.ContentAbout>
  </SectionClipPathEffect>
);

export default AboutDescription;
