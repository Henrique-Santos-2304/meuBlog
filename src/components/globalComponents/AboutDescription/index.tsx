import Headings from "components/globalComponents/Headings";
import SectionClipPathEffect from "components/globalComponents/SectionClipPathEffect";
import * as S from "./styles";
type aboutProps = {
  title: string;
  descriprions: string;
  mt?: string;
};
const AboutDescription = ({ title, descriprions, mt }: aboutProps) => (
  <SectionClipPathEffect mt={mt}>
    <S.ContentAbout>
      <Headings
        msg={title}
        colorText="#1e1e1e"
        colorEffect="#486afe"
        fontSizeText="1.8rem"
        marginText="0 0 1rem 1.5rem"
        aria-label="Titulo da seção"
      />
      <S.TextDescription aria-label="Texto com a descrição do Programador">
        {descriprions}
      </S.TextDescription>
    </S.ContentAbout>
  </SectionClipPathEffect>
);

export default AboutDescription;
