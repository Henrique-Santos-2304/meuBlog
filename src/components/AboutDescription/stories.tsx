import { Story, Meta } from "@storybook/react";
import AboutDescription from ".";

export default {
  title: "AboutDescription",
  component: AboutDescription,
} as Meta;

export const Basic: Story = (args) => <AboutDescription {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <AboutDescription {...args} />;
