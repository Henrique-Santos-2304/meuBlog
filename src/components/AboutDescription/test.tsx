import AboutDescription from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<AboutDescription />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<AboutDescription />);
    expect(container).toMatchSnapshot();
  });
  it("should have a title section and myDescription text ", () => {
    renderTheme(<AboutDescription />);

    const title = screen.getByRole("heading", {
      name: /um pouco sobre mim/i,
    });

    const myDescription = screen.getByLabelText(
      /Texto com a descrição do Programador/i
    );

    expect(title).toBeInTheDocument();
    expect(myDescription).toBeInTheDocument();
  });
});
