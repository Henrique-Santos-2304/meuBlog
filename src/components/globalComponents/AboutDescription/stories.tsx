import { Story, Meta } from "@storybook/react";
import AboutDescription from ".";

export default {
  title: "AboutDescription",
  component: AboutDescription,
} as Meta;
const mock = {
  title: "seção1",
  descriprions: "Descrição da seção",
};
export const Basic: Story = () => <AboutDescription {...mock} />;

export const Default: Story = () => <AboutDescription {...mock} />;
