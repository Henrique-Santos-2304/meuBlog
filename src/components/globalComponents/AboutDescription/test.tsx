import AboutDescription from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

const mock = {
  title: "seção1",
  descriprions: "Descrição da seção",
};
describe("<AboutDescription />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<AboutDescription {...mock} />);
    expect(container).toMatchSnapshot();
  });
  it("should have a title section and myDescription text ", () => {
    renderTheme(<AboutDescription {...mock} />);

    const title = screen.getByRole("heading", {
      name: /seção1/i,
    });

    const myDescription = screen.getByLabelText(
      /Texto com a descrição do Programador/i
    );

    expect(title).toBeInTheDocument();
    expect(myDescription).toBeInTheDocument();
  });
});
