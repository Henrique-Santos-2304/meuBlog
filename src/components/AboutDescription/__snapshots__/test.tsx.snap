// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutDescription /> should render  1`] = `
.c2 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1e1e1e;
  padding: 0;
  margin: 0 0 1rem 1.5rem;
  font-size: 2rem;
  font-family: "Roboto",monospace;
}

.c2::before {
  content: "";
  position: absolute;
  top: 50%;
  left: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.4rem;
  height: 90%;
  border-radius: 2rem;
  background: #486afe;
}

.c0 {
  width: 100vw;
  padding: 5rem 1rem;
  background: white;
  margin-top: -7rem;
  -webkit-clip-path: polygon(0 0,100% 15%,100% 100%,0 85%);
  clip-path: polygon(0 0,100% 15%,100% 100%,0 85%);
}

.c1 {
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  padding: 1.8rem 0;
}

.c3 {
  margin-top: 1rem;
  font-size: 1.6rem;
  font-weight: 500;
}

.c3 strong {
  font-style: italic;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Um pouco sobre mim
      </h1>
      <p
        aria-label="Texto com a descrição do Programador"
        class="c3"
      >
        Sou um Desenvolvedor a princípio Front End, com foco em
        <strong>
           React | Next.js
        </strong>
        , apaixonado pela Programação estou em constante aprendizado e desenvolvimento profissional sempre me atualizando, no momento crio projetos Pesssoais e freelances voltados a Desenvolvimento Web, mas ao mesmo tempo estudando sobre desenvolvimento Mobile com 
        <strong>
          React Native
        </strong>
         .
      </p>
    </div>
  </div>
</div>
`;
